releasenotes/notes/pilot-discovery-scoped-namespaces.yaml

```diff
@@ -0,0 +1,26 @@
+apiVersion: release-notes/v2
+kind: feature
+area: traffic-management
+
+# issue is a list of GitHub issues resolved in this note.
+# If issue is not in the current repo, specify its full URL instead.
+issue:
+  - https://github.com/istio/istio/issues/26679
+
+# releaseNotes is a markdown listing of any user facing changes. This will appear in the
+# release notes.
+releaseNotes:
+- |
+  **Added** An option for dynamically restricting the set of namespaces for Services, Pods, and Endpoints that istiod processes when pushing xDS updates to improve performance on the data plane.
+
+# upgradeNotes is a markdown listing of any changes that will affect the upgrade
+# process. This will appear in the release notes.
+upgradeNotes:
```

Comment:
Any consideration for upgrade?  I believe this is not enabled by default, but if it is enabled, users would need to ensure the discoverySelectors config is correct else apps could be broken.

Reply:
this is indeed disabled by default—should I still include an upgrade note describing usage of discoverySelectors?

Reply:
Since it is disabled, no.  Would be good to have it in your docs on consideration when enabling this. :)